Algorithm/QCAlgorithm.cs

```diff
@@ -1207,24 +1207,22 @@ public void SetStartDate(DateTime start)
             // no need to set this value in live mode, will be set using the current time.
             if (_liveMode) return;
 
-            //Validate the start date:
-            //1. Check range;
-            if (start < (new DateTime(1900, 01, 01)))
+            //1. Round down
+            start = start.RoundDown(TimeSpan.FromDays(1));
+
+            //2. Validate the start date range:
+            if (start < new DateTime(1900, 01, 01))
             {
                 throw new Exception("Please select a start date after January 1st, 1900.");
             }
-
-            //2. Check end date greater:
-            if (_endDate != new DateTime())
+            else if (start > DateTime.Today)
```

Comment:
Maybe today should be specified in algorithm time zone?
`todayInAlgorithmTimeZone = DateTime.UtcNow.ConvertFromUtc(TimeZone).Date;`

Reply:
I thought of that, but I think that we are dealing with algorithm timezone until we set the algorithm date/time in this method a couple of lines below:
```csharp
SetDateTime(_startDate.ConvertToUtc(TimeZone));
```

Reply:
Exactly, `_startDate` is in algorithm time zone and we shouldn't compare it with `DateTime.Today` which is in local system time zone -- think about local LEAN users in different countries around the world 😃